openss7-0.9.2.D.SLES9_K2.6.5_rc1(1)/openss7-0.9.2.D.SLES9_K2.6.5_rc1/LiS/man/man8/streams.8.man
'\" rtp
.\" -*- nroff -*- vim: ft=nroff nocin nosi
.\"
.\" @(#) streams.8.man,v 1.1.4.5 2006/04/05 10:05:01 brian Exp
.\"
.\" =========================================================================
.\"
.\" Copyright (c) 2001-2006  OpenSS7 Corporation <http://www.openss7.com/>
.\"
.\" All Rights Reserved.
.\"
.\" Permission is granted to make and distribute verbatim copies of this
.\" manual provided the copyright notice and this permission notice are
.\" preserved on all copies.
.\"
.\" Permission is granted to copy and distribute modified versions of this
.\" manual under the conditions for verbatim copying, provided that the
.\" entire resulting derived work is distributed under the terms of a
.\" permission notice identical to this one
.\" 
.\" Since the Linux kernel and libraries are constantly changing, this
.\" manual page may be incorrect or out-of-date.  The author(s) assume no
.\" responsibility for errors or omissions, or for damages resulting from
.\" the use of the information contained herein.  The author(s) may not
.\" have taken the same level of care in the production of this manual,
.\" which is licensed free of charge, as they might when working
.\" professionally.  The author(s) will take no responsibility in it.
.\" 
.\" Formatted or processed versions of this manual, if unaccompanied by
.\" the source, must acknowledge the copyright and authors of this work.
.\"
.\" -------------------------------------------------------------------------
.\"
.\" U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software
.\" on behalf of the U.S. Government ("Government"), the following
.\" provisions apply to you.  If the Software is supplied by the Department
.\" of Defense ("DoD"), it is classified as "Commercial Computer Software"
.\" under paragraph 252.227-7014 of the DoD Supplement to the Federal
.\" Acquisition Regulations ("DFARS") (or any successor regulations) and the
.\" Government is acquiring only the license rights granted herein (the
.\" license rights customarily provided to non-Government users).  If the
.\" Software is supplied to any unit or agency of the Government other than
.\" DoD, it is classified as "Restricted Computer Software" and the
.\" Government's rights in the Software are defined in paragraph 52.227-19
.\" of the Federal Acquisition Regulations ("FAR") (or any successor
.\" regulations) or, in the cases of NASA, in paragraph 18.52.227-86 of the
.\" NASA Supplement to the FAR (or any successor regulations).
.\"
.\" =========================================================================
.\" 
.\" Commercial licensing and support of this software is available from
.\" OpenSS7 Corporation at a fee.  See http://www.openss7.com/
.\" 
.\" =========================================================================
.\"
.\" Last Modified 2006/04/05 10:05:01 by brian
.\"
.\" -------------------------------------------------------------------------
.\"
.\" streams.8.man,v
.\" Revision 1.1.4.5  2006/04/05 10:05:01  brian
.\" - get references in right place
.\"
.\" Revision 1.1.4.4  2006/04/05 09:42:41  brian
.\" - moved .so line and updated headers
.\"
.\" =========================================================================
.R1
bracket-label "\fR[\fB" "\fR]" "\fR, \fB"
no-default-database
database lis.refs
accumulate
move-punctuation
abbreviate A
join-authors ", " ", " " and "
et-al " et al" 2 3
abbreviate-label-ranges ".."
sort-adjacent-labels
.R2
.so lis.macros
.\"
.\"
.TH STREAMS 8 "@PACKAGE_DATE@" "@PACKAGE@-@VERSION@" "@PACKAGE_TITLE@ System Admin."
.SH NAME
.B streams
\- incomplete man page
.\"
.\"
.SH SYNOPSIS
.PP
.HP 12
\fBstreams\fR [\fIcommand\fR]
.HP 12
\fBstreams\fR [\fIoptions\fR]
.HP 12
\fBstreams\fR {\fB-h\fR,\fB--help\fR}
.HP 12
\fBstreams\fR {\fB-V\fR,\fB--version\fR}
.HP 12
\fBstreams\fR {\fB--copying\fR}
.PD
.\"
.\"
.SH DESCRIPTION
This manpage is under construction.
.\"
.\"
.SH COMANDS
.PP
.TP
.B start
starts the
.I STREAMS
subsystem
.TP
.B stop
stops the
.I STREAMS
subsystem
.TP
.B status
reports the status of the
.I STREAMS
subsystem
.\"
.\"
.SH OPTIONS
.PP
.TP
\fB-c\fR, \fB--maxmsg\fR [\fIKBYTES\fR]
print or set the maximum message memory usage for LiS.  The default value is zero.
.TP
\fB-C\fR, \fB--maxmem\fR \fIKBYTES\fR
print or set the maximum memory usage for LiS.  The default value is zero.
.TP
\fB-s\fR, \fB--memstats\fR
prints the
.I STREAMS
memory statistics.
.TP
\fB-S\fR, \fB--qrunstats\fR
prints the
.I STREAMS
queue run statistics.
.TP
\fB-t\fR, \fB--timestats\fR
prints the
.I STREAMS
timing statistics.
.TP
\fB-m\fR, \fB--dumpmem\fR
dumps
.I STREAMS
memory allocation to the syslog file.
.TP
\fB-p\fR, \fB--dumpspl\fR
dumps SPL trace buffer to the syslog file.
.TP
\fB-q\fR, \fB--dumpqueues\fR
dumps all queues to the syslog file.
.TP
\fB-d\fR, \fB--debug1\fR \fIMASK\fR
sets the first debug mask.  The default
.I MASK
is zero.
.I MASK
is a value representing a bitwise OR of any of the following values:
.RS
.PP
.TS
tab(:);
rllsss.
0x00000001:OPEN:T{
T}
0x00000002:CLOSE:T{
T}
0x00000004:READ:T{
T}
0x00000008:WRITE:T{
T}
0x00000010:IOCTL:T{
T}
0x00000020:PUTNEXT:T{
T}
0x00000040:STRRPUT:T{
T}
0x00000080:SIG:T{
T}
0x00000100:PUTMSG:T{
T}
0x00000200:GETMSG:T{
T}
0x00000400:POLL:T{
T}
0x00000800:LINK:T{
T}
0x00001000:MEAS_TIME:T{
T}
0x00002000:MEM_LEAK:T{
T}
0x00004000:FLUSH:T{
T}
0x00008000:FATTACH:T{
T}
0x00010000:SAFE:T{
T}
0x00020000:TRCE_MSG:T{
T}
0x00040000:CLEAN_MSG:T{
T}
0x00080000:SPL_TRACE:T{
T}
0x00100000:MP_ALLOC:T{
T}
0x00200000:MP_FREEMSG:T{
T}
0x00400000:MALLOC:T{
T}
0x00800000:MONITOR_MEM:T{
T}
0x01000000:DMP_QUEUE:T{
T}
0x02000000:DMP_MBLK:T{
T}
0x04000000:DMP_DBLK:T{
T}
0x08000000:DMP_STRHD:T{
T}
0x10000000:VOPEN:T{
T}
0x20000000:VCLOSE:T{
T}
0x40000000:-:T{
T}
0x80000000:ADDRS:T{
T}
.TE
.RE
.TP
\fB-D\fR, \fB--debug2\fR \fIMASK2\fR
sets the second debug mask.  The default
.I MASK2
is zero.
.I MASK2
is a value representing a bitwise OR of any of the following values:
.RS
.PP
.TS
tab(:);
rllsss.
0x00000001:SNDFD:T{
T}
0x00000002:CP:T{
T}
0x00000004:CACHE:T{
T}
.TE
.RE
.TP
\fB-\fI<FLAG>\fR, \fB--\fI<FLAG>\fR
sets a debug flag, where
.I <FLAG>
is one of the symbolic
.B --debug1
or
.B --debug2
flag names listed above.  These options can occur multiples times.  Multiple flags are effectively OR'ed
together.
.TP
\fB-Q\fR, \fB--quiet\fR
suppresses normal output.
.TP
\fB-h\fR, \fB--help\fR, \fB-H\fR (obsolete)
print usage information and exit.
.TP
\fB-V\fR, \fB--version\fR
print version and exit.
.TP
\fB--copying\fR
print copying permissions and exit.
.\"
.\"
.SH DIAGNOSTICS
none
.\"
.\"
.SH BUGS
.PP
.B streams
has no known bugs.
.\"
.\"
.SH EXAMPLES
.PP
None.
.\"
.\"
.SH "SEE ALSO"
.PP
.BR STREAMS (4).
.\"
.\"
.SH COMPATIBILITY
.PP
.BR streams
is compatible with
.B LiS\c
.
.[
LiS
.]
.\"
.\"
.SH CONFORMANCE
.PP
.BR streams
does not conform to any standard.
.\"
.\"
.SH VERSIONS
.PP
This manpage was written for
.B LiS\c
.
.[
LiS
.]
.[
$LIST$
.]
.TI


openss7-0.9.2.D.SLES9_K2.6.5_rc1(1)/openss7-0.9.2.D.SLES9_K2.6.5_rc1/LiS/util/init/streams.sh
#!/bin/sh
#
# @(#) streams.sh,v OpenSS7-0_9_2_D_rc2(1.1.2.6) 2006/05/08 03:34:52
# Copyright (c) 2001-2006  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
# All Rights Reserved.
#
# Distributed by OpenSS7 Corporation.  See the bottom of this script for copying
# permissions.
#
# These are arguments to update-rc.d ala chkconfig and lsb.  They are recognized
# by openss7 install_initd and remove_initd scripts.  Each line specifies
# arguments to add and remove links after the the name argument:
#
# streams:	start and stop streams subsystem
# update-rc.d:	start 33 S . stop 33 0 6 .
# config:	/etc/default/streams
# probe:	false
# hide:		false
# license:	GPL
# description:	This STREAMS init script is part of Linux Fast-STREAMS.  \
#		It is responsible for ensuring that the necessary STREAMS \
#		character devices are present in the /dev directory and \
#		that the STREAMS subsystem is configured and loaded.
#

PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
name='streams'
config="/etc/default/$name"
desc="the STREAMS subsystem"

[ -e /proc/modules ] || exit 0

for STREAMS_MKNOD in /sbin/strmakenodes /usr/sbin/strmakenodes /bin/strmakenodes /usr/bin/strmakenodes ; do
    if [ -x $STREAMS_MKNOD ] ; then
	break
    else
	STREAMS_MKNOD=
    fi
done

# Specify defaults

#STREAMS_MODULES="streams streams-liskmod streams-mtdrv streams-pipemod"
STREAMS_MODULES="streams"
STREAMS_MAKEDEVICES="no"
STREAMS_REMOVEDEVICES="no"
STREAMS_MOUNTSPECFS="yes"
STREAMS_MOUNTPOINT="/dev/streams"

# Source config file
for file in $config ; do
    [ -f $file ] && . $file
done

[ -z "$STREAMS_MKNOD" ] && STREAMS_MAKEDEVICES='no'
[ -z "$STREAMS_MKNOD" ] && STREAMS_REMOVEDEVICES='no'

RETVAL=0

umask 077

if [ "${VERBOSE:-0}" -ne 0 ] ; then
    redir='>/dev/null 2>&1'
else
    redir=
fi

build_options() {
    # Build up the options string
    :
}

start() {
    echo -n "Loading STREAMS kernel modules: "
    for module in $STREAMS_MODULES ; do
	if ! grep "^$module"'[[:space:]]' /proc/modules $redir ; then
	    echo -n "$module "
	    modprobe -k -q -- $module $redir
	    [ $? -eq 0 ] || echo -n "(failed)"
	fi
    done
    echo "."

    echo -n "Starting $desc: $name "
    build_options
    RETVAL=$?
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    if grep '^[[:space:]]*streams[/.]' /etc/sysctl.conf $redir ; then
	echo -n "Reconfiguring kernel parameters: "
	sysctl -p /etc/sysctl.conf $redir
	RETVAL=$?
	if [ $RETVAL -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi
    if [ -f /etc/streams.conf ] ; then
	echo -n "Configuring STREAMS parameters: "
	sysctl -p /etc/streams.conf $redir
	RETVAL=$?
	if [ $RETVAL -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi
    if [ -n "$STREAMS_MKNOD" -a ":$STREAMS_MAKEDEVICES" = ":yes" ] ; then
	echo -n "Making STREAMS devices: "
	$STREAMS_MKNOD
	RETVAL=$?
	if [ $RETVAL -eq 0 ] ; then
	    echo "."
	else
	    echo "(failed.)"
	fi
    fi
    return $RETVAL
}

remove_modules() {
    modules=
    while read -a module ; do
	modules="${modules}${modules:+ }${module[0]}"
    done
    if [ -n "$modules" ] ; then
	echo -n "Removing STREAMS modules: "
	rmmod $modules
	RETVAL=$?
    fi
    return $RETVAL
}

stop() {
    echo -n "Stopping $desc: $name "
    RETVAL=$?
    if [ -n "$STREAMS_MKNOD" -a ":$STREAMS_REMOVEDEVICES" = ":yes" ] ; then
	echo -n "Removing STREAMS devices: "
	$STREAMS_MKNOD --remove
	RETVAL=$?
    fi
    [ $RETVAL -eq 0 ] && egrep '^streams[-_]?' /proc/modules 2>/dev/null | remove_modules
    RETVAL=$?
    if [ $RETVAL -eq 0 ] ; then
	echo "."
    else
	echo "(failed.)"
    fi
    return $RETVAL
}

restart() {
    stop
    start
    return $?
}

show() {
    echo "$name.sh: show: not yet implemented." >&2
    return 1
}

usage() {
    echo "Usage: /etc/init.d/$name.sh (start|stop|restart|force-reload|show)" >&2
    return 1
}

case "$1" in
    (start|stop|restart|show)
	$1 || RETVAL=$?
	;;
    (force-reload)
	restart || RETVAL=$?
	;;
    (*)
	usage || RETVAL=$?
	;;
esac

[ "${0##*/}" = "$name.sh" ] && exit $RETVAL

# =============================================================================
# 
# @(#) streams.sh,v OpenSS7-0_9_2_D_rc2(1.1.2.6) 2006/05/08 03:34:52
#
# -----------------------------------------------------------------------------
#
# Copyright (c) 2001-2006  OpenSS7 Corporation <http://www.openss7.com>
# Copyright (c) 1997-2000  Brian F. G. Bidulock <bidulock@openss7.org>
#
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or modify it under
# the terms of the GNU General Public License as published by the Free Software
# Foundation; version 2 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
# details.
#
# You should have received a copy of the GNU General Public License along with
# this program; if not, write to the Free Software Foundation, Inc., 675 Mass
# Ave, Cambridge, MA 02139, USA.
#
# -----------------------------------------------------------------------------
#
# U.S. GOVERNMENT RESTRICTED RIGHTS.  If you are licensing this Software on
# behalf of the U.S. Government ("Government"), the following provisions apply
# to you.  If the Software is supplied by the Department of Defense ("DoD"), it
# is classified as "Commercial Computer Software" under paragraph 252.227-7014
# of the DoD Supplement to the Federal Acquisition Regulations ("DFARS") (or any
# successor regulations) and the Government is acquiring only the license rights
# granted herein (the license rights customarily provided to non-Government
# users).  If the Software is supplied to any unit or agency of the Government
# other than DoD, it is classified as "Restricted Computer Software" and the
# Government's rights in the Software are defined in paragraph 52.227-19 of the
# Federal Acquisition Regulations ("FAR") (or any successor regulations) or, in
# the cases of NASA, in paragraph 18.52.227-86 of the NASA Supplement to the FAR
# (or any successor regulations).
#
# -----------------------------------------------------------------------------
#
# Commercial licensing and support of this software is available from OpenSS7
# Corporation at a fee.  See http://www.openss7.com/
#
# -----------------------------------------------------------------------------
#
# Last Modified 2006/05/08 03:34:52 by brian
#
# =============================================================================

# vim: ft=sh sw=4 tw=80
